### PR TITLE
Feature/initial testing

### PR DESCRIPTION
**Pull request by colefiscus**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [ ] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [x] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
This PR adds some initial Cypress tests for our application. The components/features being tested include the pages appearance, the proper functioning of the date input and the loading of a new picture based on that input, the Vue Router links, and if the links in the About section function correctly.

## *Instructions for reviewer:*
The reviewer should look in the new test files.

## *Issues:*
These tests are passing as the app is written at the time of creation of this branch. The tests will need minor updates later and more tests will be needed as features are added.

One other issue is there is a GET request that is failing that I think has to do with Vue? I believe it is a compiler error, but it only seems to error when running Cypress and it doesn't actually affect the tests.

### CODE REVIEW

File: cypress/fixtures/potd_data.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+
+  "date":"2021-02-27",
+
+  "explanation":"Sometimes it do be like that.",
```
Comment (richardltyler) [inline cypress/fixtures/potd_data.json:5]: You know it do.

File: cypress/integration/about-spec.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      .get("footer").get("a:first").click()
+  })
+
+  it("Should show a description of the site when going to About page", () => {
+    cy.get("main").get("article").get("h2").contains("🚀 👽 About the Vueniverse 👽 🚀").should("be.visible")
+  })
+
+  it("Should contain links to the API used and the inspiration for the site", () => {
+    cy
+      .get("main").get("article").get("p:last").get("a:first").should("have.attr", "href", "https://apod.nasa.gov/apod/astropix.html")
```
Comment (lbmerchant93) [inline cypress/integration/about-spec.js:15]: Digging the "have.attr" here!

Comment (richardltyler) [inline cypress/integration/about-spec.js:15]: Good job testing the links here

File: cypress/integration/click-to-enter-spec.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  it("Should contain a blast-off button and additional text", () => {
+    cy
+      .get("main").get("button").contains("🤠 CLICK HERE TO BLAST OFF 🤠").should("be.visible")
+      .get("main").get("p").contains("(WARNING FLASHING LIGHTS!!)").should("be.visible")
+  })
+
+  it("Should contain a footer with component links", () => {
+    cy
+      .get("footer").get("a:first").contains("About").should("be.visible")
+      .get("footer").get("a:last").contains("Home").should("be.visible")
```
Comment (richardltyler) [inline cypress/integration/click-to-enter-spec.js:23]: These are going to change soon so lets keep that in mind for later. The `home` link is going to be in the header

File: cypress/integration/picture-of-the-day-spec.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  it("Should load a picture to display when navigating to the home page", () => {
+    cy
+      .get("footer").get("a:last").click()
+      .get("img").should("have.attr", "src", "https://apod.nasa.gov/apod/image/2102/PIA24333_fig1_1035c.jpg")
+  })
+
+  it("Should display a new picture after changing the date input", () => {
+    cy
+      .get("footer").get("a:last").click()
+      .get("header").get("input").type("2020-12-25", "{enter}").trigger("change").wait(500)
```
Comment (richardltyler) [inline cypress/integration/picture-of-the-day-spec.js:30]: This is a work of art my friend

File: cypress/integration/routes-spec.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+describe("Vue Routes", () => {
+
+  beforeEach(() => {
+    cy.visit("http://localhost:8080")
+  })
+  
+  it("Should visit intial component on page load", () => {
+    cy.url().should("include", "8080/")
```
Comment (richardltyler) [inline cypress/integration/routes-spec.js:8]: Really good idea testing that the link only includes the slash there

**Review by lbmerchant93 — APPROVED** [review]
Looks great! Will have to do some final tweaking when the finished styling is done but this is definitely an awesome start! Great job on stubbing the two calls and intercepting with two fixtures. Amazing job getting cypress to do the date input! I had no clue how to do that and appreciate you figuring it out and sharing the knowledge!

**Review by richardltyler — APPROVED** [review]
Looks good!

**Pull request merged**